File: mockez-front/src/app/core/service/project.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Project } from '@core/model/project';
import { environment } from '@environment/environment';

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

@Injectable({
  providedIn: 'root'
})
export class ProjectService {

  constructor(
    private httpClient: HttpClient
  ) {
  }

  getProject(id: string): Observable<Project> {
    return this.httpClient.get<Project>(`${environment.apiUrl}/project/${id}`);
  }

  getProjects(): Observable<Project[]> {
    return this.httpClient.get<Project[]>(`${environment.apiUrl}/project`);
  }

  saveOrUpdate(project: Project): Observable<string> {
    return this.httpClient.post<string>(`${environment.apiUrl}/project`, project);
  }

  deleteProject(id: string): Observable<string> {
    return this.httpClient.delete<string>(`${environment.apiUrl}/project/${id}`);
  }
}

File: mockez-back/src/main/java/com/mocker/service/ProjectService.java
package com.mocker.service;

import com.mocker.domain.model.entity.Project;

import java.util.List;
import java.util.UUID;

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

public interface ProjectService {

    Project getProject(UUID id);

    List<Project> getProjects();

    UUID saveOrUpdate(Project project);

    UUID delete(UUID id);

}
